dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/compute/config/DimensionDataComputeServiceContextModule.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.compute.config;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Provides;
+import org.jclouds.collect.Memoized;
+import org.jclouds.compute.config.ComputeServiceAdapterContextModule;
+import org.jclouds.dimensiondata.cloudcontrol.DimensionDataCloudControlApi;
+import org.jclouds.dimensiondata.cloudcontrol.compute.suppliers.OrganisationIdForAccount;
+import org.jclouds.dimensiondata.cloudcontrol.domain.BaseImage;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Datacenter;
+import org.jclouds.dimensiondata.cloudcontrol.domain.internal.ServerWithExternalIp;
+import org.jclouds.rest.AuthorizationException;
+import org.jclouds.rest.suppliers.MemoizedRetryOnTimeOutButNotOnAuthorizationExceptionSupplier;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.jclouds.Constants.PROPERTY_SESSION_INTERVAL;
+
+public class DimensionDataComputeServiceContextModule
+      extends ComputeServiceAdapterContextModule<ServerWithExternalIp, BaseImage, BaseImage, Datacenter> {
+
+   @Override
+   protected void configure() {
+
+      install(new DimensionDataComputeServiceDependenciesModule());
+
+      // to have the compute service adapter override default locations
+      install(new LocationsFromComputeServiceAdapterModule<ServerWithExternalIp, BaseImage, BaseImage, Datacenter>() {
+      });
+
+      super.configure();
+   }
+
+   @Provides
+   @Singleton
+   @Memoized
+   public final Supplier<String> getOrganisationIdForAccount(AtomicReference<AuthorizationException> authException,
+         @Named(PROPERTY_SESSION_INTERVAL) long seconds, DimensionDataCloudControlApi api) {
+      return MemoizedRetryOnTimeOutButNotOnAuthorizationExceptionSupplier
+            .create(authException, new OrganisationIdForAccount(api), seconds, TimeUnit.SECONDS);
+   }
```

Comment:
Better move this supplier to the HttpApi module, since it is generic for the provider APIs, not specifically about the compute abstraction.